pkg/types/credentials.go

```diff
@@ -58,24 +58,53 @@ func (c *Credentials) MarshalJSON() ([]byte, error) {
 
 // Validate implements InputValidator and verifies all mandatory fields are populated
 func (c *Credentials) Validate() error {
-	if c.Basic != nil {
-		if c.Basic.Username == "" {
-			return errors.New("missing broker username")
+	hasCertificate := len(httpclient.GetHttpClientGlobalSettings().ServerCertificate) > 0
+	if hasCertificate {
+		if c.Basic != nil {
+			//when there is tls, its ok to username and password will be empty
+			//throw an error only if there are incomplete credentials
+			if c.Basic.Username != "" && c.Basic.Password == "" {
+				return errors.New("missing broker password")
+			}
+			if c.Basic.Password != "" && c.Basic.Username == "" {
+				return errors.New("missing broker username")
+			}
 		}
-		if c.Basic.Password == "" {
-			return errors.New("missing broker password")
+		if c.TLS != nil {
+			if c.TLS.Certificate != "" && c.TLS.Key == "" {
+				return errors.New("missing certificate")
+			}
+			if c.TLS.Key != "" && c.TLS.Certificate == "" {
+				return errors.New("missing certificate key")
+			}
+			if c.TLS.Key != "" && c.TLS.Certificate != "" {
+				_, err := tls.X509KeyPair([]byte(c.TLS.Certificate), []byte(c.TLS.Key))
+				if err != nil {
+					return errors.New("invalidate TLS configuration: " + err.Error())
+				}
+			}
+
+		}
+	} else {
+		if c.Basic != nil {
```

Comment:
i think you missed some cases:
1. user is switching from basic to tls when sm dose not have tls
2. user is switching from tls to basic when sm dose not have tls